odl/test/tomo/geometry/geometry_test.py

```diff
@@ -9,102 +9,348 @@
-    assert all_equal(geom.axis, np.array([1, 0, 0]))
-    geom = odl.tomo.Parallel3dAxisGeometry(apart, dpart, axis=[0, 1, 0])
-    assert all_equal(geom.axis, np.array([0, 1, 0]))
-    geom = odl.tomo.Parallel3dAxisGeometry(apart, dpart, axis=[0, 0, 1])
-    assert all_equal(geom.axis, np.array([0, 0, 1]))
-    geom = odl.tomo.Parallel3dAxisGeometry(apart, dpart, axis=[1, 2, 3])
-    assert all_equal(geom.axis,
-                     np.array([1, 2, 3]) / np.linalg.norm([1, 2, 3]))
-
+    # Zero not allowed as axis
     with pytest.raises(ValueError):
-        odl.tomo.Parallel3dAxisGeometry(apart, dpart, axis=(1,))
+        odl.tomo.Parallel3dAxisGeometry(apart, dpart, axis=[0, 0, 0])
+
+    # Detector axex should not be parallel or otherwise result in a
+    # linear dependent triplet
     with pytest.raises(ValueError):
-        odl.tomo.Parallel3dAxisGeometry(apart, dpart, axis=(1, 2))
+        odl.tomo.Parallel3dAxisGeometry(
+            apart, dpart, det_axes_init=([0, 1, 0], [0, 1, 0]))
     with pytest.raises(ValueError):
-        odl.tomo.Parallel3dAxisGeometry(apart, dpart, axis=(1, 2, 3, 4))
+        odl.tomo.Parallel3dAxisGeometry(
+            apart, dpart, det_axes_init=([0, 0, 0], [0, 1, 0]))
+
+    # check str and repr work without crashing and return a non-empty string
+    assert str(geom) > ''
+    assert repr(geom) > ''
 
-    # check str and repr work without crashing and return something
-    assert str(geom)
-    assert repr(geom)
+
+def test_parallel_3d_orientation(axis):
+    """Check if the orientation is positive for any ``axis``."""
+    full_angle = np.pi
+    apart = odl.uniform_partition(0, full_angle, 10)
+    dpart = odl.uniform_partition([0, 0], [1, 1], (10, 10))
+    geom = odl.tomo.Parallel3dAxisGeometry(apart, dpart, axis=axis)
+
+    norm_axis = np.array(axis, dtype=float) / np.linalg.norm(axis)
+    assert all_almost_equal(geom.axis, norm_axis)
+    # The symmetry axis should be the second detector axis by default
+    assert all_almost_equal(geom.axis, geom.det_axes_init[1])
+
+    # The detector-to-source normal should always be perpendicular to both
+    # detector axes, and the triplet (normal, det_ax0, det_ax1) should form
+    # a positively oriented system
+    for angle in [0, np.pi / 2, 3 * np.pi / 4, np.pi]:
+        dot0 = np.dot(geom.det_to_src(angle, [0, 0]), geom.det_axes(angle)[0])
+        dot1 = np.dot(geom.det_to_src(angle, [0, 0]), geom.det_axes(angle)[1])
+        assert dot0 == pytest.approx(0)
+        assert dot1 == pytest.approx(0)
+
+        normal_det_to_src = (geom.det_to_src(angle, [0, 0]) /
+                             np.linalg.norm(geom.det_to_src(angle, [0, 0])))
+
+        axis_0, axis_1 = geom.det_axes(angle)
+        orient = np.linalg.det(np.vstack([normal_det_to_src, axis_0, axis_1]))
+        assert orient == pytest.approx(1)
+
+    # check str and repr work without crashing and return a non-empty string
+    assert str(geom) > ''
```

Comment:
Why is that stronger? It only checks that `bool(str(geom)) == True`, while `str(geom) > ''` will fail if the left operand is not string-like (e.g. `None`), and it will evaluate to `False` if the string is empty (OK, not sure why that would happen).